x-pack/plugins/security_solution/public/management/pages/policy/view/policy_advanced.tsx

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useCallback } from 'react';
+import { useDispatch } from 'react-redux';
+import { EuiFieldText, EuiFormRow, EuiPanel, EuiText } from '@elastic/eui';
+import { cloneDeep } from 'lodash';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { policyConfig } from '../store/policy_details/selectors';
+import { usePolicyDetailsSelector } from './policy_hooks';
+import { AdvancedPolicySchema } from '../models/advanced_policy_schema';
+
+function setValue(obj: Record<string, unknown>, value: string, path: string[]) {
+  let newPolicyConfig = obj;
+  for (let i = 0; i < path.length - 1; i++) {
+    if (!newPolicyConfig[path[i]]) {
+      newPolicyConfig[path[i]] = {} as Record<string, unknown>;
+    }
+    newPolicyConfig = newPolicyConfig[path[i]] as Record<string, unknown>;
+  }
+  newPolicyConfig[path[path.length - 1]] = value;
+}
+
+function getValue(obj: Record<string, unknown>, path: string[]) {
```

Comment:
you shoudl add a return type to this function (`string` right?)

Reply:
you can add both types to the function's return signature:

```suggestion
	function getValue(obj: Record<string, unknown>, path: string[]): string | undefined {
```

Reply:
actually since `currentPolicyConfig` is type `Record<string, unknown>`, L37 is returning type `unknown`; should i be casting to a `string`? how does the return type help here?

Reply:
@paul-tavares